Composer/packages/client/src/App.tsx

```diff
@@ -42,6 +43,10 @@ export const App: React.FC = () => {
     ipcRenderer?.on('cleanup', (_event) => {
       performAppCleanupOnQuit();
     });
+
+    ipcRenderer?.on('machine-info', (_event, info) => {
+      setMachineInfo(info);
```

Comment:
It looks like you are using the machine info only inside the useSurveyNotification hook. If so, why do we need to create a recoil state for it. You can just use the `ipcRenderer.on` inside the hook. Ex `useInitializeLoger.ts`

Reply:
I was trying to make this work last week, but it runs into annoying race conditions where the message handler isn't initialized before the message gets sent and so nothing gets through. Using the Recoil store lets us put the machine info down somewhere whenever we receive it and then count on it being available when we need it.

Reply:
Makes sense to usethe recoil state unless the hook can be registered in app.tsx

Reply:
Can you explain the race condition a bit more. I'm also confused why we can't just initialize the survey here when the event is received? If the message handler doesn't get initialized before the message is sent, we would never get the message, right?